ssl/statem/statem_dtls.c

```diff
@@ -44,14 +44,39 @@ static const unsigned char bitmask_start_values[] =
 static const unsigned char bitmask_end_values[] =
     { 0xff, 0x01, 0x03, 0x07, 0x0f, 0x1f, 0x3f, 0x7f };
 
-static void dtls1_set_message_header_int(SSL_CONNECTION *s, unsigned char mt,
-                                         size_t len,
-                                         unsigned short seq_num,
-                                         size_t frag_off,
-                                         size_t frag_len);
 static int dtls_get_reassembled_message(SSL_CONNECTION *s, int *errtype,
                                         size_t *len);
 
+static dtls_sent_msg *dtls1_sent_msg_new(size_t msg_len)
+{
+    dtls_sent_msg *msg = NULL;
+    unsigned char *msg_buf = NULL;
+
+    if ((msg = OPENSSL_zalloc(sizeof(*msg))) == NULL)
+        return NULL;
+
+    if (msg_len) {
+        if ((msg_buf = OPENSSL_malloc(msg_len)) == NULL) {
```

Comment:
As an optimization, this could be done as a single alloc.

Reply:
I'm not sure how? Could you provide an example?